exe/facter.cc

```diff
@@ -264,6 +264,9 @@ int main(int argc, char **argv)
         bool show_legacy = vm.count("show-legacy");
         facts.write(boost::nowide::cout, fmt, queries, show_legacy);
         boost::nowide::cout << endl;
+    } catch (locale_error const& e) {
+        cerr << "Failed to initialize logging system due to a locale error: " << e.what() << endl;
```

Comment:
The colorize flag is set very last thing in setup_l;ogging. Since that fails when locale is bad, we don't get around to enabling colorized output.

I don't think it's worth trying to rejigger colorization initialization for what is likely to be an extremely rare code path. The normal warning code path when we only fail once is still colorized properly